ambuda/views/auth.py

```diff
@@ -79,51 +91,96 @@ def _is_valid_reset_token(row: db.PasswordResetToken, raw_token: str, now=None):
     return True
 
 
+# Copied from https://wtforms.readthedocs.io/en/2.3.x/validators/
```

Comment:
hm - if we already have the `Length` validator, why roll our own validator here?

Reply:
> hm - if we already have the `Length` validator, why roll our own validator here?

When I used the native val.Length() validator, the sign in form silently snips `username` and `password` to the maximum length. Database stores the snipped `username` and `password`. 

For instance, user may have copy pasted 240 chars our form will only store 64 bytes of it to the db. Creating our own validator will show a clear error that username cannot exceed 64 bytes. I'm not sure if it is a bug or a feature in the val.Length().